### DISCUSSION TITLE
Reducing size of libraries

### DISCUSSION THREAD

**Discussion by sekinan** [emgucv/emgucv#463] [General]
Any ideas for reducing the size of the libraries, using in a C# WPF application on Net5. The distributed application is enormous, over 2.5GB just for the OpenCV/EmguCV elements (cudnn_cnn_infer64_8.dll alone is 644,156KB). We don't need any of the neural networking and inference engine stuff. Is there any way of removing them from the nuget?

**Comment by emgucv** [comment]
You can use the Emgu.CV.runtime.windows nuget package (no cuda, ~48MB):
https://www.nuget.org/packages/Emgu.CV.runtime.windows/ 

instead of the Emgu.CV.runtime.windows.cuda nuget package (with cuda, 2.5GB with dependency. CUDA binary are huge.):
https://www.nuget.org/packages/Emgu.CV.runtime.windows.cuda/

**Reply by sekinan** [reply]
Thanks, but I should have been more clear. We need some of the CUDA functionality for performance reasons (BlockMatch for example)

**Reply by emgucv** [reply]
In that case, you will need the full CUDA package including all the CUDA & CUDNN dlls. 

Another options is to build Emgu CV with CUDA package but without CUDNN, that way you don't need to deploy the CUDNN dlls. It is not a configuration we support at this moment. Please feel free to give it a try.

**Reply by sekinan** [reply]
Thanks for the swift reply. It was what I was fearing, but no matter, I'll give it a go. Thanks again
